tests/functional/api/BackbeatServer.js

```diff
@@ -235,6 +272,333 @@ describe('Backbeat Server', () => {
             });
         });
 
+        it('should get a 200 response for route: /_/crr/failed', done => {
+            const body = JSON.stringify([{
+                bucket: 'bucket',
+                key: 'key',
+                versionId: 'versionId',
+                site: 'site',
+            }]);
+            makeRetryPOSTRequest(body, (err, res) => {
+                assert.ifError(err);
+                assert.strictEqual(res.statusCode, 200);
+                done();
+            });
+        });
+
+        const invalidPOSTRequestBodies = [
+            'a',
+            {},
+            [],
+            ['a'],
+            [{
+                // Missing bucket property.
+                key: 'b',
+                versionId: 'c',
+                site: 'd',
+            }],
+            [{
+                // Missing key property.
+                bucket: 'a',
+                versionId: 'c',
+                site: 'd',
+            }],
+            [{
+                // Missing versionId property.
+                bucket: 'a',
+                key: 'b',
+                site: 'd',
+            }],
+            [{
+                // Missing site property.
+                bucket: 'a',
+                key: 'b',
+                versionId: 'c',
+            }],
+        ];
+
+        invalidPOSTRequestBodies.forEach(body => {
+            const invalidBody = JSON.stringify(body);
+            it('should get a 400 response for route: /_/crr/failed when ' +
```

Comment:
You should also include at least one non-JSON body to test

Reply:
I believe the `'a'` POST body test fails the JSON parsing.